### DISCUSSION TITLE
Pixels

### DISCUSSION THREAD

**Discussion by Mayank4884** [openPupil/Open-PupilEXT#32] [General]
Even if I am Changing the image width and height in the pupil settings.. Why there is no change in the pupil diameter?

**Comment by BZandi** [comment]
If I remember correctly, the specified pupil diameter in pixels is always relative to the original resolution. The image height and width in the pupil settings should not be changed as these belongs to the pupil algorithms. In my opinion, It is better to use the ROI feature for reducing the computation time.

As the PupilEXT project is open source you can check out the code:
[https://github.com/openPupil/Open-PupilEXT/tree/main/src/pupil-detection-methods](https://github.com/openPupil/Open-PupilEXT/tree/main/src/pupil-detection-methods)

Further, it is recommended to read the publications of the pupil detection algorithms if you are interested how they work:

Dongheng Li and Derrick J. Parkhurst. Starburst: A robust algorithm for video-based eye tracking. IEEE Computer Society Conference on Computer Vision and Pattern Recognition (CVPR), September 2005.

Lech Swirski, Andreas Bulling, and Neil Dodgson. Robust real-time pupil tracking in highly off-axis images. In Proceedings - 2012 ACM Symposium on Eye Tracking Research and Applications (ETRA), pages 173–176, 2012.

Wolfgang Fuhl, Thomas Kübler, Katrin Sippel, Wolfgang Rosenstiel, and Enkelejda Kasneci. Excuse: robust pupil detection in real-world scenarios. In International Conference on Computer Analysis of Images and Patterns, pages 39–51. Springer, 2015.

Wolfgang Fuhl, Thiago C. Santini, Thomas Kübler, and Enkelejda Kasneci. ElSe: Ellipse selection for robust pupil detection in real-world environments. In Proceedings - 2016 ACM Symposium on Eye Tracking Research and Applications (ETRA), volume 14, pages 123–130, 2016.

Thiago Santini, Wolfgang Fuhl, and Enkelejda Kasneci. PuRe: Robust pupil detection for real-time pervasive eye tracking. Computer Vision and Image Understanding, 170:40–50, 2018.

Thiago Santini, Wolfgang Fuhl, and Enkelejda Kasneci. PuReST: Robust pupil tracking for real-time pervasive eye tracking. In Proceedings - 2018 ACM Symposium on Eye Tracking Research and Applications (ETRA). ACM, 2018.

Thiago Santini, Wolfgang Fuhl, David Geisler and Enkelejda Kasneci. EyeRecToo: Open-source Software for Real-time Pervasive Head-mounted Eye Tracking. VISIGRAPP 2017.

**Comment by Mayank4884** [comment]
If I take 2 images with different working distance but of same pupil size and upload it as externally recorded image (offline).. It is giving different result (both pupil size appearing different) .. What should I do?

**Reply by BZandi** [reply]
This is an expected behaviour as you use a single camera and the pixel-size of the circular object changes with different distances to the camera. When using a stereo-camera setup, the size is provided in mm and the object's size should remain the same with different distances to the camera system. The question was already answered in another thread. Follow this link: https://github.com/openPupil/Open-PupilEXT/discussions/14
